static/js/components/CommentTree.js

```diff
@@ -50,104 +57,123 @@ const renderComment = R.curry(
     forms: FormsState,
     upvote: CommentVoteFunc,
     downvote: CommentVoteFunc,
+    loadMoreComments: LoadMoreCommentsFunc,
     beginEditing: (fk: string, iv: Object, e: ?Object) => void,
     processing: boolean,
     depth: number,
-    comment: Comment
+    comment: GenericComment
   ) => {
-    const formKey = replyToCommentKey(comment)
-    const editFormKey = editCommentKey(comment)
-    const initialValue = getCommentReplyInitialValue(comment)
+    if (comment.comment_type === "comment") {
```

Comment:
Could you refactor this to separate functions for rendering each of the comment types?

Reply:
`"comment"` should also be refactored to a `const`

Reply:
`"comment"` actually needs to be a literal here for flow to derive the appropriate type of `comment`. I could move this into a const within the same file, but flow assumes consts exported from other modules are not really const. What do you think?

Reply:
ok, fair enough, I think the tradeoff of having flow be able to assert correctness here is more important than having `const`s